src/components/pages/import.js

```diff
@@ -0,0 +1,158 @@
+import by from 'thenby';
+import Page from '../page.js';
+import PropTypes from 'prop-types';
+import urls from '../../utils/urls.js';
+import React, { useState } from 'react';
+import Logger from '../../common/logger.js';
+import { useTranslation } from 'react-i18next';
+import { Tooltip, Button, Table, Select, Spin } from 'antd';
+import { inject, useService } from '../container-context.js';
+import { DOCUMENT_IMPORT_TYPE } from '../../common/constants.js';
+import ImportApiClient from '../../services/import-api-client.js';
+import { useDateFormat, useLanguage } from '../language-context.js';
+import LanguageNameProvider from '../../data/language-name-provider.js';
+import CountryFlagAndName from '../localization/country-flag-and-name.js';
+import { CloudDownloadOutlined, CloudSyncOutlined } from '@ant-design/icons';
+
+const logger = new Logger(import.meta.url);
+
+function Import({ initialState, importApiClient }) {
+  const { t } = useTranslation('import');
+
+  const { language } = useLanguage();
+  const { formatDate } = useDateFormat();
+  const languageNameProvider = useService(LanguageNameProvider);
+
+  const sourceMenuItems = initialState.importSources;
+  const [selectedSource, setSelectedSource] = useState(null);
+  const [selectedDocumentKeys, setSelectedDocumentKeys] = useState([]);
+  const [importableDocuments, setImportableDocuments] = useState([]);
+
+  const handleImportClick = () => {
+    const tasks = selectedDocumentKeys
+      .map(key => importableDocuments.find(doc => doc.key === key))
+      .map(doc => ({
+        key: doc.key,
+        importedRevision: doc.importedRevision,
+        importableRevision: doc.importableRevision
+      }));
+    logger.info(`Dummy import sets tasks: ${JSON.stringify(tasks)}`);
+  };
+
+  const handleSourceMenuChange = async value => {
+    const newSelectedSource = sourceMenuItems.find(source => source.name === value);
+    setSelectedSource(newSelectedSource);
+    setImportableDocuments([]);
+    setSelectedDocumentKeys([]);
+
+    const { documents } = await importApiClient.getImports(newSelectedSource);
+    setImportableDocuments(documents);
+  };
+
+  const renderImportType = doc => {
+    let icon = doc.importType;
```

Comment:
Does it make sense to set the icon (which should be a ReactElement) to the enum value? We could just initialize it with `null`instead.

Reply:
I considered the case when there would be an extra import type and we'd forget to map it, we'd at least see the text. Crazy, now come to think of it, the level of anticipation 😄